docs/docs/markers.mdx

```diff
@@ -0,0 +1,291 @@
+
+### Marker Configuration
+
+To define one or more marker, you need to specify a `yaml` configuration file. Here is an example:
+
+```yaml
+marker_name_provided:
+    slot_was_set: name
+marker_mood_expressed:
+    or:
+        - intent: mood_unhappy
+        - intent: mood_great
+marker_cheer_up:
+    seq:
+        - intent: mood_unhappy
+        - action: utter_cheer_up
+marker_bot_challenged:
+    at_least_once:
+        - intent: bot_challenge
+marker_mood_expressed_and_name_provided:
+    and:
+        - slot_was_set: name
+        - or:
+            - intent: mood_unhappy
+            - intent: mood_great
+```
+
+The top level contains unique user-specified names for each marker. This is a dictionary mapping marker names to marker definitions.
+
+Each marker definition can contain an arbitrary combination of `conditions` and `operators` that map to a valid sub-marker definition.
+
+Each `condition` maps a condition tag to a value. For example, in `marker_name_provided`, the tag `slot_was_set` is mapped to the slot `name` which exists in the `domain.yaml` file.
+
+Each `operator` is a dictionary mapping an operator tag (e.g., `and`) to a list of sub markers, which can be conditions or a nested operator-conditions definition (e.g., see `marker_mood_expressed_and_name_provided`).
+
+:::note
+You cannot yet reuse an existing marker name in the definition of another marker.
+
+:::
+
+## Running the Markers script
+
+Markers are extracted from the dialogues already stored in your [tracker store](https://rasa.com/docs/rasa/next/tracker-stores). Once you've created your Marker definitions in the Makers configuration file, you can extract them by running the follow command:
+
+```bash
+rasa evaluate markers markers.yaml extracted_markers.csv all
+```
+
+The argument `all` will cause the script to process all the trackers in your tracker store. To process a subset you can use a different tracker selection strategy, either `first_n` or `sample`. Refer to the CLI usage for details.
+
+Each tracker in the tracker store can contain multiple sessions identified by a `sender_id`. When extracting markers and computing their statistics, each session is considered separately.
+
+
+### Extracted Markers
+
+We extract information about top-level markers which have user-specified names, and extract meta information about all the events where the markers applied.
+
+For the `at_least_once` marker we only extract meta information about the first occurrence.
+
+We extract the following meta information:
+
+1. the number of user turns (i.e., the number of `UserUttered` events) preceding the event at which a marker applied
+2. the index of the event at which a marker applied
+
+We output the extracted information in a `.csv` file that contains the following columns:
+
+- sender_id
+- session_idx
+- marker name
+- event_idx (where we index all events in the tracker, starting with index 0)
+- num_preceeding_user_turns
```

Comment:
The `.csv` file below shows all the column names already, so if we prefer to keep this portion I might suggest using it as a legend where each column name is followed by what the user can expect from the values. 

This may also be a great place to explain what `nan` values may be expected.